system/solaris_zone.py

```diff
@@ -0,0 +1,360 @@
+        self.msg            = []
+
+        self.module         = module
+        self.path           = self.module.params['path']
+        self.name           = self.module.params['name']
+        self.whole_root     = self.module.params['whole_root']
+        self.root_password  = self.module.params['root_password']
+        self.timeout        = self.module.params['timeout']
+        self.config         = self.module.params['config']
+        self.create_options = self.module.params['create_options']
+
+        self.zoneadm_cmd    = self.module.get_bin_path('zoneadm', True)
+        self.zonecfg_cmd    = self.module.get_bin_path('zonecfg', True)
+        self.ssh_keygen_cmd = self.module.get_bin_path('ssh-keygen', True)
+
+    def create(self):
+        if not self.path:
+            self.module.fail_json(msg='Missing required argument: path')
+
+        t = tempfile.NamedTemporaryFile(delete = False)
+
+        if self.whole_root:
+            t.write('create -b %s\n' % self.create_options)
+            self.msg.append('creating whole root zone')
+        else:
+            t.write('create %s\n' % self.create_options)
+            self.msg.append('creating sparse root zone')
+
+        t.write('set zonepath=%s\n' % self.path)
+
+        if self.config:
+            for line in self.config:
+                t.write('%s\n' % line)
+        t.close()
+
+        cmd = '%s -z %s -f %s' % (self.zonecfg_cmd, self.name, t.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to create zone. %s' % (out + err))
+        os.unlink(t.name)
+
+        cmd = '%s -z %s install' % (self.zoneadm_cmd, self.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to install zone. %s' % (out + err))
+
+        self.configure_sysid()
+        self.configure_password()
+        self.configure_ssh_keys()
+
+    def configure_sysid(self):
+        os.unlink('%s/root/etc/.UNCONFIGURED' % self.path)
+
+        open('%s/root/noautoshutdown' % self.path, 'w').close()
+
+        node = open('%s/root/etc/nodename' % self.path, 'w')
+        node.write(self.name)
+        node.close
+
+        id = open('%s/root/etc/.sysIDtool.state' % self.path, 'w')
+        id.write('1       # System previously configured?\n')
+        id.write('1       # Bootparams succeeded?\n')
+        id.write('1       # System is on a network?\n')
+        id.write('1       # Extended network information gathered?\n')
+        id.write('0       # Autobinder succeeded?\n')
+        id.write('1       # Network has subnets?\n')
+        id.write('1       # root password prompted for?\n')
+        id.write('1       # locale and term prompted for?\n')
+        id.write('1       # security policy in place\n')
+        id.write('1       # NFSv4 domain configured\n')
+        id.write('0       # Auto Registration Configured\n')
+        id.write('vt100')
```

Comment:
Should the terminal type be configurable?  Would `xterm` be a better default than `vt100`, nowadays?